lntypes/channel_party.go

```diff
@@ -50,3 +50,76 @@ func (p ChannelParty) IsLocal() bool {
 func (p ChannelParty) IsRemote() bool {
 	return p == Remote
 }
+
+// Dual represents a structure when we are tracking the same parameter for both
+// the Local and Remote parties.
+type Dual[A any] struct {
+	// Local is the value tracked for the Local ChannelParty
+	Local A
+
+	// Remote is the value tracked for the Remote ChannelParty
+	Remote A
+}
+
+// GetParty gives Dual an access method that takes a ChannelParty as an
+// argument. It is included for ergonomics in cases where the ChannelParty is
+// in a variable and which party determines how we want to access the Dual.
+func (d *Dual[A]) GetParty(p ChannelParty) A {
```

Comment:
How about `GetForParty()` and `SetForParty()`? Since we're not really setting the party but the value for a party.